iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/variant.inl

```diff
@@ -175,7 +177,7 @@ inline bool variant<Types...>::emplace_at_index(CTorArguments&&... args) noexcep
     using T = typename internal::get_type_at_index<0, TypeIndex, Types...>::type;
 
     call_element_destructor();
-    new (m_storage) T(std::forward<CTorArguments>(args)...);
+    new (&m_storage.data) T(std::forward<CTorArguments>(args)...);
```

Comment:
Why not just `new (&m_storage) ...`?

Reply:
I also think this should work if data is the only member, at least here where `void*` is expected.

Reply:
@elBoberido I don't know, maybe some copy/paste mistake. I removed `.data`.